File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counts = defaultdict(int)
        for x in nums:
            counts[x] += 1

        arr = []
        for num, count in counts.items():
            arr.append([count, num])
        arr.sort()

        output = []
        while len(output) < k:
            output.append(arr.pop()[1])
        return output

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array-ii/submission-2.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int l = 0, r = 0;
        while (r < nums.size()) {
            int count = 1;
            while (r + 1 < nums.size() && nums[r] == nums[r + 1]) {
                count++;
                r++;
            }

            for (int i = 0; i < min(count, 2); i++) {
                nums[l] = nums[r];
                l++;
            }
            r++;
        }
        return l;
    }
};